openmc/plotter.py

```diff
@@ -0,0 +1,910 @@
+                                  xs[name][line](energy_grid))
+
+    return energy_grid, data
+
+
+def calculate_mgxs(this, data_type, types, orders=None, temperature=294.,
+                   cross_sections=None, ce_cross_sections=None,
+                   enrichment=None):
+    """Calculates multi-group cross sections of a requested type.
+
+    If the data for the nuclide or macroscopic object in the library is
+    represented as angle-dependent data then this method will return the
+    geometric average cross section over all angles.
+
+    Parameters
+    ----------
+    this : str or openmc.Material
+        Object to source data from
+    data_type : {'nuclide', 'element', material', 'macroscopic'}
+        Type of object to plot
+    types : Iterable of values of PLOT_TYPES_MGXS
+        The type of cross sections to calculate
+    orders : Iterable of Integral, optional
+        The scattering order or delayed group index to use for the
+        corresponding entry in types. Defaults to the 0th order for scattering
+        and the total delayed neutron data.
+    temperature : float, optional
+        Temperature in Kelvin to plot. If not specified, a default
+        temperature of 294K will be plotted. Note that the nearest
+        temperature in the library for each nuclide will be used as opposed
+        to using any interpolation.
+    cross_sections : str, optional
+        Location of MGXS HDF5 Library file. Default is None.
+    ce_cross_sections : str, optional
+        Location of continuous-energy cross_sections.xml file. Default is None.
+        This is used only for expanding an openmc.Element object passed as this
+    enrichment : float, optional
+        Enrichment for U235 in weight percent. For example, input 4.95 for
+        4.95 weight percent enriched U. Default is None
+        (natural composition).
+
+    Returns
+    -------
+    energy_grid : numpy.ndarray
+        Energies at which cross sections are calculated, in units of eV
+    data : numpy.ndarray
+        Cross sections calculated at the energy grid described by energy_grid
+
+    """
+
+    # Check types
+    cv.check_type('temperature', temperature, Real)
+    if enrichment:
+        cv.check_type('enrichment', enrichment, Real)
+    cv.check_iterable_type('types', types, str)
+
+    cv.check_type("cross_sections", cross_sections, str)
+    library = openmc.MGXSLibrary.from_hdf5(cross_sections)
+
+    if data_type in ('nuclide', 'macroscopic'):
+        mgxs = _calculate_mgxs_nuc_macro(this, types, library, orders,
+                                         temperature)
+    elif data_type in ('element', 'material'):
+        mgxs = _calculate_mgxs_elem_mat(this, types, library, orders,
+                                        temperature, ce_cross_sections,
+                                        enrichment)
+    else:
+        raise TypeError("Invalid type")
+
+    # Convert the data to the format needed
```

Comment:
Optional: Would be more readable to add some color to the final form of `energy_grid` rather than indicating that there is a needed formatting change.